tables/Tables/test/tTableUtil.cc

```diff
@@ -0,0 +1,276 @@
+  }
+  // Replace the subtable.
+  {
+    TableDesc td;
+    td.addColumn(ScalarColumnDesc<String>("col1"));
+    td.addColumn(ScalarColumnDesc<String>("col3"));
+    Table tab(TableUtil::createTable("tTableUtil_tmp/maindata::SubTab2::SubSubTab6",
+                                     td, Table::New));
+    AlwaysAssertExit (tab.nrow() == 0);
+    AlwaysAssertExit (tab.tableDesc().ncolumn() == 2);
+    AlwaysAssertExit (tab.tableDesc()[0].name() == "col1");
+    AlwaysAssertExit (tab.tableDesc()[1].name() == "col3");
+  }
+}
+
+void testErrors()
+{
+  // Do some erroneous createTable calls.
+  TableDesc td;
+  Bool ok = False;
+  try {
+    // SubTab22 does not exist
+    TableUtil::createTable ("tTableUtil_tmp/maindata::SubTab22::SubSubTab4",
+                            td, Table::New);
+  } catch (const TableError& x) {
+    ok = True;
+    cout << "Expected exception: " << x.what() << endl;
+  }
+  AlwaysAssertExit (ok);
+  ok = False;
+  try {
+    // main33data does not exist.
+    TableUtil::createTable ("tTableUtil_tmp/main33data::SubTab2::SubSubTab4",
+                            td, Table::NewNoReplace);
+  } catch (const TableError& x) {
+    ok = True;
+    cout << "Expected exception: " << x.what() << endl;
+  }
+  AlwaysAssertExit (ok);
+  ok = False;
+  try {
+    // SubSubTab4 already exists.
+    TableUtil::createTable ("tTableUtil_tmp/maindata::SubTab2::SubSubTab4",
+                            td, Table::NewNoReplace);
+  } catch (const TableError& x) {
+    ok = True;
+    cout << "Expected exception: " << x.what() << endl;
+  }
+  AlwaysAssertExit (ok);
+  ok = False;
+  try {
+    // Empty part given.
+    TableUtil::createTable ("tTableUtil_tmp/main3data::::SubTab2::SubSubTab4",
+                            td, Table::New);
+  } catch (const TableError& x) {
+    ok = True;
+    cout << "Expected exception: " << x.what() << endl;
+  }
+  AlwaysAssertExit (ok);
+}    
+
+void testLayoutInfo (const String& tableName)
+{
+  cout << endl << "Test getLayout and tableInfo ..." << endl;
+  // Get the description and #rows of the Table.
+  TableDesc layout;
+  cout << "TableUtil::getlayout #rows = "
+       << TableUtil::getLayout (layout, tableName);
+  layout.show (cout);
+  cout << endl;
+  TableInfo info(TableUtil::tableInfo (tableName));
+  cout << "type = " << info.type() << endl;
+  cout << "subtype = " << info.subType() << endl;
+  cout << info.readme() << endl;
+}
+
+int main()
+{
+  try {
+    // Create subdirectory with this name, which will be deleted by assay.
+    Directory dir("tTableUtil_tmp");
+    dir.create();
+    createTables();
+    readTables();
+    testColons();
+    testDelete();
+    testReplace();
+    testErrors();
+    testLayoutInfo("tTableUtil_tmp/maindata");
+    testLayoutInfo("tTableUtil_tmp/maindata::SubTab2");
+  } catch (const std::exception& x) {
+    cout << "Caught an exception : " << x.what() << endl;
+    return 1;
+  } 
+  cout << "OK" << endl;
```

Comment:
I rather see something more explicit, since anything could have written "OK".
```suggestion
  cout << "tTableUtil: All tests passed." << endl;
```